runelite-mixins/src/main/java/net/runelite/mixins/RSNPCMixin.java

```diff
@@ -110,6 +111,13 @@ public void onCompositionChanged(RSNPCComposition composition)
 		}
 	}
 
+	@FieldHook("composition")
+	@Inject
+	public void afterCompositionChanged(int idx)
+	{
+		client.getCallbacks().post(new NpcCompositionChanged(this));
```

Comment:
new assignment can be null on npc despawn

Reply:
I don't understand why this would be a problem

Reply:
It fires the event for comp change, when it is really a npc despawn. Seems bad?

Reply:
Well technically the composition does change when it despawns. I don't have a strong opinion about it though so I guess I'll change it